### PR TITLE
feat: .NET Functions Static File support

### PR DESCRIPTION
**Pull request by peterinnesmsft**
### Purpose
For skills-based scenarios, we need to support serving static files from the .NET Functions-based bots, for content such as manifest, .lu and .qna files. This PR adds an HTTP trigger out of the box to support this scenario.

### Changes
- Update host.json to change the http route prefix to "" from the default of "api".
- Updating MessagesTrigger and SkillsTrigger to add "api" to routes.
- Adding StaticFilesTrigger to handle returning of supported static files.
- Update scaffolded .csproj to include all files within 'www' directory as part of build drop.

### Tests
Manually tested via local Functions app, as well as published Functions app on Azure.

### CODE REVIEW

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            if (!TryGetFilePath(context, path, out string filePath) ||
+                !_provider.TryGetContentType(filePath, out string contentType))
+            {
+                return new NotFoundResult();
+            }
+
+            return new FileStreamResult(new FileStream(filePath, FileMode.Open), contentType);
+        }
+
+        bool TryGetFilePath(ExecutionContext context, string relativePath, out string result)
```
Comment (thomas-chung) [inline generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:41]: nit: private

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.StaticFiles;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+
+namespace <%= botName %>.Triggers
+{
+    public class StaticFilesTrigger
+    {
+        private const string StaticFilesDirectory = "www";
```
Comment (thomas-chung) [inline generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:13]: Should this be "wwwroot" instead so it's inline with the static file version counterpart in the web app version?

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+        private readonly FileExtensionContentTypeProvider _provider;
+
+        public StaticFilesTrigger()
+        {
+            var provider = new FileExtensionContentTypeProvider();
+            provider.Mappings[".lu"] = "application/vnd.microsoft.lu";
+            provider.Mappings[".qna"] = "application/vnd.microsoft.qna";
```
Comment (thomas-chung) [inline generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:21]: What about manifest files extensions? Do we need to provide that as well?

**Review by joshgummersall — APPROVED** [review]
Looks functionally consistent with JS.

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs (hunk 2 above)
Comment (thomas-chung) [reply generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:13]: I guess the flip side is on Azure Function the web app seems to be on wwwroot already so it's like wwwroot/wwwroot.

**Review by thomas-chung — APPROVED** [review]

**Review by tracyboehrer — APPROVED** [review]

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs (hunk 3 above)
Comment (peterinnesmsft) [reply generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:21]: FileExtensionContentTypeProvider contains a default mapping for many different file extensions to corresponding content types. As the schema files are just .json, these are already covered by the default mapping. The way this code is written just supplements that by adding the additional ones needed for our scenarios.

File: generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs (hunk 2 above)
Comment (peterinnesmsft) [reply generators/generator-bot-adaptive/generators/app/templates/dotnet/functions/Triggers/StaticFilesTrigger.cs:13]: Yeah, it's a little weird for sure. Looking at other implementations in places, this seems to be the way it's been handled. At the end of the day, customers can easily change it to something a little more fitting should they wish.

**Pull request merged**